<launch>
    <arg name="manager_name"            default="following_control_nodelet_manager" />
    <arg name="manager_threads"         default="4" />

    <arg name="rviz"                    default="false"/>
    <arg name="rviz_cfg"                default="$(find person_following_control)/config/rviz/person_following_control.rviz"/>
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_cfg)"/>

    <arg name="rqt_reconfigure"         default="true"/>
    <arg name="following_method"        default="0" />
    <!--# 0 : virtual_spring_model_dynamic_window_approach
        # 1 : virtual_spring_model
        # 2 : dynamic_window_approach -->
    <arg name="use_smoother"        default="false" />

    <group ns = "person_following_control">
        <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen">
            <param name="num_worker_threads" value="$(arg manager_threads)" />
        </node>

        <node pkg="nodelet" type="nodelet" name="person_following_control" args="load person_following_control/PersonFollowing $(arg manager_name)" >
            <rosparam file="$(find person_following_control)/param/virtual_spring_model_dynamic_window_approach_param.yaml" command="load" />
            <param name="following_method"      value="$(arg following_method)" />

            <remap if="$(arg use_smoother)"         from="cmd_vel" to="/velocity_smoother/raw_cmd_vel"/>
            <remap unless="$(arg use_smoother)"     from="cmd_vel" to="/cmd_vel_mux/input/teleop"/>
        </node>

        <include if="$(arg use_smoother)" file="$(find person_following_control)/launch/velocity_smoother.launch">
            <param name="manager_name" value="$(arg manager_name)" />
        </include>

    </group>
    <node if="$(arg rqt_reconfigure)" pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure"/>
</launch>